tracing-error/src/backtrace.rs

```diff
@@ -0,0 +1,65 @@
+use super::try_bool;
+use crate::layer::WithContext;
+use std::fmt;
+use tracing::{Metadata, Span};
+
+#[derive(Clone)]
+pub struct SpanBacktrace {
+    span: Span,
+}
+
+// === impl SpanBacktrace ===
+
+impl SpanBacktrace {
+    pub fn capture() -> Self {
+        SpanBacktrace {
+            span: Span::current(),
+        }
+    }
+
+    pub fn with_spans(&self, f: impl FnMut(&'static Metadata<'static>, &str) -> bool) {
+        self.span.with_subscriber(|(id, s)| {
+            if let Some(getcx) = s.downcast_ref::<WithContext>() {
+                getcx.with_context(s, id, f);
+            }
+        });
+    }
+}
+
+impl fmt::Debug for SpanBacktrace {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let mut err = None;
+        let mut span = 0;
+
+        writeln!(f, "span backtrace:")?;
```

Comment:
i would probably omit this and let the error type/whatever is formatting the span trace add whatever kind of header etc it wants?

Reply:
I put this in to be consistent with the way that std::backtrace::Backtrace works, I would prefer to have people who want to do custom formatting do so via the `with_spans` API